tests/conftest.py

```diff
@@ -46,3 +57,28 @@ def testbed(request):
 
     tbinfo = TestbedInfo(tbfile)
     return tbinfo.testbed_topo[tbname]
+
+@pytest.fixture(autouse=True)
+def loganalyzer(duthost, request):
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="loganalyzer")
+    # Add start marker into DUT syslog
+    marker = loganalyzer.init()
+    yield loganalyzer
+    if not request.config.getoption("--disable_loganalyzer") and not pytest.disable_loganalyzer:
+        # Read existed common regular expressions located with legacy loganalyzer module
+        loganalyzer.load_common_config()
+        result = loganalyzer.analyze(marker)
+        if not result:
+            pytest.fail("Log analyzer failed.")
```

Comment:
If we fail in this condition why don't we put an end marker anyway?

Reply:
End marker is added by "analyze" method.